src/CommonDBTM.php

```diff
@@ -1521,6 +1521,10 @@ public function update(array $input, $history = 1, $options = [])
                 $this->oldvalues = [];
 
                 foreach (array_keys($this->input) as $key) {
+                    if (is_array($this->input[$key])) {
+                        continue;
+                    }
```

Comment:
I do not like this one. Unless I am wrong, it cannot trigger a warning unless the field exists in database, so why ignoring it ?

Reply:
I have the following warning if I disable it:
```
[2022-01-17 11:04:38] glpiphplog.NOTICE:   *** PHP Notice (8): Array to string conversion in /home/adrie/localhost/gmaster/src/DBmysql.php at line 1263
  Backtrace :
  src/DBmysql.php:1393                               DBmysql::quoteValue()
  src/DBmysql.php:1427                               DBmysql->buildUpdate()
  src/CommonDBTM.php:608                             DBmysql->update()
  src/CommonDBTM.php:1597                            CommonDBTM->updateInDB()
  ...ncedforms/front/targets/emailtarget.form.php:55 CommonDBTM->update()
  ...ncedforms/front/targets/emailtarget.form.php:48 updateTarget()
  
```

Reply:
The problem is that `if (array_key_exists($key, $this->fields)) {` is evaluated to true due to the fact that you recreate this "virtual" field in `post_getFromDB()` method of your item.

As update should only handle real db fields, I think the fix should be to do this
```diff
diff --git a/src/CommonDBTM.php b/src/CommonDBTM.php
index 90fc95df29..607d22079d 100644
--- a/src/CommonDBTM.php
+++ b/src/CommonDBTM.php
@@ -1521,7 +1521,7 @@ class CommonDBTM extends CommonGLPI
                 $this->oldvalues = [];
 
                 foreach (array_keys($this->input) as $key) {
-                    if (array_key_exists($key, $this->fields)) {
+                    if ($DB->fieldExists(static::getTable(), $key)) {
                       // Prevent history for date statement (for date for example)
                         if (
                             is_null($this->fields[$key])
```

Reply:
👍  done.